### PR TITLE
Feat : New user section added

### PR DESCRIPTION
**Pull request by sahsisunny**
# Feat : New user section added

## Description

- In this PR, I have added a new user section in the home page using the new user component.

## files changed: local or package?

- I have added a new three folders
  - In `src` folder, I have added a new folder `components` and in that folder, I have added a new folder `new-member` and in that folder, I have added a new file `index.tsx` and in that file, I have added a new component `NewMembersCard`.
 - And `newMemberCard` component is added in `index.tsx` file.
  - In `public` folder, I have added a new folder `images` and in that folder, I have added a new png file `Avatar.png`.
- Also add `./src/**/*.{js,ts,jsx,tsx}` in `tailwind.config.js` file to make the tailwind work properly.

## Screenshots
New User Section

### CODE REVIEW

File: src/components/new-member/index.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import React from "react";
+import Image from "next/image";
+
+const NewMembersCard = ({
+  newMemberName,
+  newMemberImageSrc,
+}: {
+  newMemberName: string;
```
Comment (shmbajaj) [inline src/components/new-member/index.tsx:8]: Write type separately, like this
```ts
type NewMembersCardProps = {
newMemberName: string;
newMemberImageSrc: string;
}
```
Then can use like this
```tsx
const NewMembersCard = ({
  newMemberName,
  newMemberImageSrc,
}:NewMembersCardProps
```

File: src/components/new-member/index.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  newMemberImageSrc: string;
+}) => {
+  return (
+    <div className="flex bg-white max-w-sm rounded-md transition-shadow duration-300 h-35 overflow-hidden">
+      <div className="flex-col p-4 cursor-default flex-1 rounded-md border-0 shadow-none">
```
Comment (shmbajaj) [inline src/components/new-member/index.tsx:13]: - Remove default css properties: cursor-default, border-0 and shadow-none.
- Check if image and heading needs to be centerd.

File: src/components/new-member/index.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        <div>
+          <Image
+            src={newMemberImageSrc}
+            className="w-12 h-12 rounded-full mx-auto visible-visible opacity-1"
+            alt="Avatar"
+          />
+        </div>
+        <h2 className="text-sm py-2 font-bold font-roboto">{newMemberName}</h2>
```
Comment (shmbajaj) [inline src/components/new-member/index.tsx:21]: Either remove font-bold and text-sm or use something tag like p or use lower level of heading.
Also remove font-roboto.

**Review by shmbajaj — CHANGES_REQUESTED** [review]
- Check your css, validate it and remove unnecessary code
- Use absolute import
- Write tests
- Don't hardcode response in pages/index.tsx
-

File: pages/index.tsx [hunk 4]
```diff
@@ -54,6 +59,19 @@ export default function Home() {
         </div>
       </main>
 
+      <div>
+        <h1 className="text-center text-2xl font-bold">New Users</h1>
+        <div className="flex flex-wrap w-full mx-auto justify-center">
+          {Array.from({ length: NUM_MEMBERS }).map((_, i) => (
+            <NewMembersCard
+              key={i}
```
Comment (satyam73) [inline pages/index.tsx:67]: Key shouldn't be index here

**Review by satyam73 — CHANGES_REQUESTED** [review]

File: pages/index.tsx [hunk 5]
```diff
@@ -1,6 +1,11 @@
-import Head from 'next/head'
-import Image from 'next/image'
-import styles from '../styles/Home.module.css'
+import Head from "next/head";
+import Image from "next/image";
+import styles from "../styles/Home.module.css";
+import NewMembersCard from "../src/components/new-member";
+import React from "react";
```
Comment (Neha) [inline pages/index.tsx:5]: Should move to the top (before styles)

File: pages/index.tsx [hunk 6]
```diff
@@ -54,6 +59,19 @@ export default function Home() {
         </div>
       </main>
 
+      <div>
+        <h1 className="text-center text-2xl font-bold">New Users</h1>
```
Comment (Neha) [inline pages/index.tsx:63]: Move the content to constant file

File: src/components/new-member/index.tsx [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+  newMemberImageSrc: string;
+}) => {
+  return (
+    <div className="flex bg-white max-w-sm rounded-md transition-shadow duration-300 h-35 overflow-hidden">
+      <div className="flex-col p-4 cursor-default flex-1 rounded-md border-0 shadow-none">
+        <div>
+          <Image
+            src={newMemberImageSrc}
+            className="w-12 h-12 rounded-full mx-auto visible-visible opacity-1"
+            alt="Avatar"
```
Comment (Neha) [inline src/components/new-member/index.tsx:18]: make the alt dynamic with username

File: src/components/new-member/index.tsx (hunk 1 above)
Comment (sahsisunny) [reply src/components/new-member/index.tsx]: done

File: src/components/new-member/index.tsx (hunk 3 above)
Comment (sahsisunny) [reply src/components/new-member/index.tsx]: removed

File: pages/index.tsx (hunk 5 above)
Comment (sahsisunny) [reply pages/index.tsx]: done

File: pages/index.tsx (hunk 6 above)
Comment (sahsisunny) [reply pages/index.tsx]: done

File: src/components/new-member/index.tsx (hunk 7 above)
Comment (sahsisunny) [reply src/components/new-member/index.tsx]: done

File: src/components/new-member/index.tsx (hunk 2 above)
Comment (sahsisunny) [reply src/components/new-member/index.tsx]: removed

File: pages/index.tsx (hunk 4 above)
Comment (sahsisunny) [reply pages/index.tsx:69]: It is temporary for rendering the dummy data ,

File: pages/index.tsx [hunk 8]
```diff
@@ -1,6 +1,13 @@
...
-import Image from 'next/image'
-import styles from '../styles/Home.module.css'
+import React, { useState, useEffect } from "react";
+import Head from "next/head";
+import Image from "next/image";
+import styles from "@/styles/Home.module.css";
+import NewMembersCard from "@/src/components/new-member";
+import { NEW_USER, NUM_MEMBERS_NUMBER } from "@/src/constants/AppConstants";
+
+const avatarImageSrc = require("../public/images/avatar.png");
```
Comment (shmbajaj) [inline pages/index.tsx:8]: From next time onwards, please add a comment for informing about the temporary use of hard-coded or spoofed data.

**Review by shmbajaj — APPROVED** [review]

File: pages/index.tsx (hunk 8 above)
Comment (sahsisunny) [reply pages/index.tsx:8]: sure

File: pages/index.tsx [hunk 9]
```diff
@@ -1,6 +1,13 @@
-import Head from 'next/head'
-import Image from 'next/image'
-import styles from '../styles/Home.module.css'
+import React, { useState, useEffect } from "react";
+import Head from "next/head";
+import Image from "next/image";
+import styles from "@/styles/Home.module.css";
+import NewMembersCard from "@/src/components/new-member";
```
Comment (vvaibhavdesai) [inline pages/index.tsx:5]: Please sort imports check readme for reference

File: pages/index.tsx [hunk 10]
```diff
@@ -17,7 +24,7 @@ export default function Home() {
         </h1>
 
         <p className={styles.description}>
-          Get started by editing{' '}
+          Get started by editing{" "}
```
Comment (vvaibhavdesai) [inline pages/index.tsx:27]: what is this line here for ?

File: pages/index.tsx [hunk 11]
```diff
@@ -54,18 +61,32 @@ export default function Home() {
         </div>
       </main>
 
+      <div>
+        <h1 className="text-center text-3xl font-bold">{NEW_USER}</h1>
+        <div className="flex flex-wrap w-full mx-auto justify-center">
+          {Array.from({ length: NUM_MEMBERS_NUMBER }).map((_, i) => (
```
Comment (vvaibhavdesai) [inline pages/index.tsx:67]: just add a comment here regarding use of  `Array.from`

File: pages/index.tsx [hunk 12]
```diff
@@ -54,18 +61,32 @@ export default function Home() {
...
+      </div>
+
       <footer className={styles.footer}>
         <a
           href="https://vercel.com?utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
           target="_blank"
           rel="noopener noreferrer"
         >
-          Powered by{' '}
+          Powered by{" "}
```
Comment (vvaibhavdesai) [inline pages/index.tsx:84]: ???

File: src/components/new-member/index.tsx [hunk 13]
```diff
@@ -0,0 +1,33 @@
+import React from "react";
```
Comment (vvaibhavdesai) [inline src/components/new-member/index.tsx:1]: fix the folder name

**Review by vvaibhavdesai — COMMENTED** [review]
Great work 🚀  but needs some refactoring I have dropped comments.
If you need any help in understanding more on comments reach out to me on discord.

File: pages/index.tsx (hunk 9 above)
Comment (sahsisunny) [reply pages/index.tsx]: done

File: pages/index.tsx (hunk 10 above)
Comment (sahsisunny) [reply pages/index.tsx]: automatically change due to mine vs code settings

File: pages/index.tsx (hunk 11 above)
Comment (sahsisunny) [reply pages/index.tsx:69]: added

File: src/components/new-member/index.tsx (hunk 13 above)
Comment (sahsisunny) [reply src/components/new-member/index.tsx]: done

File: pages/index.tsx (hunk 12 above)
Comment (sahsisunny) [reply pages/index.tsx]: automatically change due to mine vs code settings

File: pages/index.tsx [hunk 14]
```diff
@@ -54,6 +60,23 @@ export default function Home() {
         </div>
       </main>
 
+      <div>
+        <h1 className="text-center text-3xl font-bold">{NEW_USER}</h1>
+        <div className="flex flex-wrap w-full mx-auto justify-center">
+          {/* 
+            I am using Array.from() to create an array of length NUM_MEMBERS_NUMBER and then mapping over it. It is temporary solution to render the "New Member" card multiple times. After the API integration, we will get the data from the API and then map over it to render the "New Member" card.
```
[the remark this answers is not in the document]
Comment (sahsisunny) [reply pages/index.tsx]: done

**Review by vvaibhavdesai — APPROVED** [review]

**Pull request merged**